### PR TITLE
Improve the module bundling story

### PR DESCRIPTION
**Pull request by gnandretta**
- [x] Stop getting the version from package.json with packageify because it needs a JSON loader when bundling with webpack.
- [x] Compile js (es2015) and jsx files with babel to use them in the package.
- [x] Remove the need for a css loader/transform.
- [x] Stop storing compiled css in the repo.
- [x] Add an example for browserify.
- [x] Add an example for webpack.
- [x] Add documentation for the examples.
- [x] Update README instructions for NPM installation.
- [x] Update deploy script to publish the compiled files instead of the original source code.

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -152,12 +172,14 @@ module.exports = function(grunt) {
...
   grunt.loadNpmTasks('grunt-contrib-watch');
   grunt.loadNpmTasks("grunt-env");
+  grunt.loadNpmTasks("grunt-exec");
   grunt.loadNpmTasks("grunt-http");
 
 
-  grunt.registerTask("build", ["clean:build", "env:build", "browserify:build", "uglify:build"]);
-  grunt.registerTask("dev", ["clean:dev", "connect:dev", "browserify:dev", "watch"]);
-  grunt.registerTask("design", ["clean:dev", "connect:dev", "browserify:design", "watch"]);
+  grunt.registerTask("build", ["clean:build", "env:build", "stylus:build", "browserify:build", "uglify:build"]);
```
Comment (daf-spr) [inline Gruntfile.js]: I know it is a "nitpicking" detail, so do if you think it worth, maybe creating a task:
```grunt.registerTask("dev-prepare", ["clean:dev", "connect:dev", "stylus:build"]);```
and then using it in "dev" and "design" so adding new tasks is easier.

File: css/index.css [hunk 2]
```diff
@@ -1,1048 +0,0 @@
-@import "normalize.css";
```
Comment (daf-spr) [inline css/index.css]: :+1:

File: examples/bundling/browserify/package.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  },
+  "author": "",
+  "license": "MIT",
+  "browserify": {
+    "transform": [
+      "babelify"
+    ]
+  },
+  "dependencies": {
+    "auth0-lock-passwordless": "file:../../../"
```
Comment (daf-spr) [inline examples/bundling/browserify/package.json]: Cool. Didn't knew this "trick"

File: .babelrc [hunk 4]
```diff
@@ -0,0 +1,4 @@
+{
+  "plugins": ["version-inline", "transform-css-import-to-string"],
```
Comment (daf-spr) [inline .babelrc]: Cool plugins :smiley:

**Comment by daf-spr** [comment]
Looks great! Assumed you used babel plugins instead of browserify / webpack ones to avoid duplication and allow adding more bundlers easily :+1:. Tested the bundling process and works ok :+1:. Give a look to my "nitpicking" comment if you want then LGTM.

**Comment by gnandretta** [comment]
@daf-spr Thanks!

**Pull request merged**